auth_interfaces.go

```diff
@@ -16,7 +16,7 @@ type User interface {
 	Username() string
 	GetAPIKey() string
 	Roles() []string
-	HasPermission(string, string, int) (bool, error)
+	HasPermission(string, string, string, int) (bool, error)
```

Comment:
This is now three string argument in a row. Can these go in a struct?

Reply:
+1

Reply:
done